components/calendar/src/astronomy.rs

```diff
@@ -605,15 +1110,43 @@ impl Astronomical {
         }
         result
     }
+    #[allow(dead_code)]
+    pub(crate) fn sine_offset(moment: Moment, location: Location, alpha: f64) -> f64 {
+        let phi = location.latitude;
+        let tee_prime = Location::universal_from_local(moment, location);
+        let delta = Self::declination(tee_prime, 0.0, Self::solar_longitude(tee_prime));
+
+        tan_degrees(phi) * tan_degrees(delta)
+            + sin_degrees(alpha) / (cos_degrees(delta) * cos_degrees(phi))
+    }
 }
 
 #[cfg(test)]
 mod tests {
 
     use super::*;
 
-    const TEST_LOWER_BOUND: f64 = 0.99999999;
-    const TEST_UPPER_BOUND: f64 = 1.00000001;
+    // Constants applied to provide a margin of error when comparing floating-point values in tests.
+    const TEST_LOWER_BOUND_FACTOR: f64 = 0.9999999;
+    const TEST_UPPER_BOUND_FACTOR: f64 = 1.0000001;
+
+    // Location of mecca from the lisp code
+    const MECCA: Location = Location {
+        latitude: 6427.0 / 300.0,
+        longitude: 11947.0 / 300.0,
+        elevation: 298.0,
+        zone: (1_f64 / 8_f64),
+    };
+
+    fn assert_eq_f64(expected_value: f64, value: f64, moment: Moment) {
```

Comment:
Suggestion (optional): if you wrote this as `macro_rules` instead of `fn`, then errors would show up with the correct line number. Instead, you'll need to generate a backtrace in order to see which exact case was failing.

Reply:
done!